### PR TITLE
[armadillo] Fix build error in Linux

### PR DESCRIPTION
**Pull request by LilyWangL**
Building `armadillo` failed in Linux:
```
CMake Error at scripts/cmake/vcpkg_fixup_cmake_targets.cmake:53 (message):

  '/home/vwangli/Lily/vcpkg/packages/armadillo_x64-linux/debug/share/armadillo/CMake'
  does not exist.
Call Stack (most recent call first):
  ports/armadillo/portfile.cmake:28 (vcpkg_fixup_cmake_targets)
  scripts/ports.cmake:71 (include)
```
Because of case sensitivity in Linux, so I modified CONFIG_PATH _armadillo/CMake_ to _Armadillo/CMake_ in **portfile.cmake**.

Related issue: #6808

### CODE REVIEW

**Comment by cenit** [comment]
Isn’t it easier to modify the `vcpkg_fixup_cmake_target` to `share/Armadillo/CMake`?

**Comment by wangli28** [comment]
@cenit Thanks for your feedback!

File: ports/armadillo/portfile.cmake [hunk 1]
```diff
@@ -25,11 +25,12 @@ vcpkg_configure_cmake(
...
 
 vcpkg_install_cmake()
-vcpkg_fixup_cmake_targets(CONFIG_PATH share/armadillo/CMake)
+vcpkg_fixup_cmake_targets(CONFIG_PATH share/Armadillo/CMake)
 
 vcpkg_copy_pdbs()
 
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/share/Armadillo)
```
Comment (cenit) [inline ports/armadillo/portfile.cmake]: On Windows/macOS (or any case-insensitive filesystem) it destroys the armadillo folder!

Comment (cenit) [reply ports/armadillo/portfile.cmake]: Sorry this line was not in the initial patch... strange that it passed CI

**Pull request merged**